File: DAMOS/trunk/tests/org.eclipselabs.damos.codegen.c.tests/src/org/eclipselabs/damos/codegen/c/gtests/MatrixScalarMultiplyTest.java
/****************************************************************************
 * Copyright (c) 2008, 2012 Andreas Unger and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *    Andreas Unger - initial API and implementation 
 ****************************************************************************/

package org.eclipselabs.damos.codegen.c.gtests;

import org.eclipselabs.damos.codegen.c.test.AbstractGeneratorGTest;
import org.eclipselabs.damos.codegen.c.test.GTest;
import org.eclipselabs.damos.dml.Block;
import org.eclipselabs.damos.dml.Outport;
import org.eclipselabs.damos.library.base.util.sources.ConstantConstants;
import org.eclipselabs.damos.mscript.util.TypeUtil;
import org.junit.Before;

/**
 * @author Andreas Unger
 *
 */
@GTest(sourceFile="gtests/MatrixScalarMultiplyTest.cpp", program="gtests/MatrixScalarMultiplyTest")
public class MatrixScalarMultiplyTest extends AbstractGeneratorGTest {

	@Before
	@Override
	public void setUp() {
		super.setUp();
		
		Block constant = createBlock(CONSTANT, "Constant");
		setSynchronousTimingConstraint(constant, 1);
		setArgument(constant, ConstantConstants.PARAMETER__VALUE, "{ { 1, 2 }, { 3, 4 }, { 5, 6 } }");
		
		Block arrayScalarMultiply = createTestBlock("MatrixScalarMultiply", "MatrixScalarMultiply");
		
		Outport outport = createOutport("Out", createArrayTypeSpecification(TypeUtil.createRealType(), 3, 2));
		
		connect(constant, arrayScalarMultiply);
		connect(arrayScalarMultiply, outport);

		generateAndCompile();
	}

}

File: DAMOS/trunk/tests/org.eclipse.damos.codegen.c.tests/gtests/MatrixScalarMultiplyTest.cpp
/*******************************************************************************
 * Copyright (c) 2008, 2012 Andreas Unger and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     Andreas Unger - initial API and implementation
 *******************************************************************************/
#include "gtest/gtest.h"
#include "MatrixScalarMultiplyTest.h"

TEST(MatrixScalarMultiplyTest, MatrixScalarMultiply) {
	Output output;

	initialize();

	execute(&output);
	EXPECT_DOUBLE_EQ(11.9, output.out[0][0]);
	EXPECT_DOUBLE_EQ(23.8, output.out[0][1]);
	EXPECT_DOUBLE_EQ(35.7, output.out[1][0]);
	EXPECT_DOUBLE_EQ(47.6, output.out[1][1]);
	EXPECT_DOUBLE_EQ(59.5, output.out[2][0]);
	EXPECT_DOUBLE_EQ(71.4, output.out[2][1]);
}
